src/domains/access.ml

```diff
@@ -455,23 +455,30 @@ let check_accs (prev_r,prev_lp,prev_w) (conf,w,loc,e,lp) =
     (new_r, new_lp, new_w)
   | _ -> (prev_r,prev_lp,prev_w)
 
-let check_safe ls (accs,lp) prev_safe =
+(* Check if two accesses race and if yes with which confidence *)
+let conflict2 (conf,mhp,w,loc,e,lp) (conf2,mhp2,w2,loc2,e2,lp2) =
+  if (not w) && (not w2) then
+    None (* two read/read accesses do not conflict *)
+  else if not (LSSet.is_empty @@ LSSet.inter lp lp2) then
+    None (* the labelled string set excludes that these conflict *)
+  else if not (MHP.may_happen_in_parallel mhp mhp2) then
+    None (* They may not be in parallel *)
+  else
+    Some (max conf conf2)
+
+let check_safe ls (accs,_) prev_safe =
```

Comment:
I now remember that this code is notably refactored in #397, which is currently only part of the interactive branch for interactive race warnings. I'll have to check exactly how big the conflicts would be, but in that form the whole `MHP.t` stuff also needs to be a `Lattice.S` to be usable in the global invariant.

Reply:
I think we're not yet at the point to merge #391 entirely, but I suppose I could merge #397 also to master now to allow those conflicts to be solved already here instead of leaving more conflicts to be merged for #391.

Reply:
Yes, I think that makes sense!

Reply:
Actually, it probably would just have to be `Printable.S`, not `Lattice.S`, since it's either only used in the variables or some set elements.